lib/models/LifecycleRule.ts

```diff
@@ -136,10 +140,16 @@ export default class LifecycleRule {
 
     /**
      * NoncurrentVersionExpiration
-     * @param days - NoncurrentDays
+     * @param prop - Property must be defined in `validProps`
+     * @param value - integer for `NoncurrentDays` and `NewerNoncurrentVersions` 
      */
-    addNCVExpiration(days: number) {
-        this.ncvExpiration = { NoncurrentDays: days };
+    addNCVExpiration(prop: 'NoncurrentDays' | 'NewerNoncurrentVersions', value: number): this;
+    addNCVExpiration(prop: string, value: number) {
+        const validProps = ['NoncurrentDays', 'NewerNoncurrentVersions'];
+        if (validProps.includes(prop)) {
+            this.ncvExpiration = this.ncvExpiration || { NoncurrentDays: 0 };
```

Comment:
Are we setting 0 days as a default because the two NCV rules combine with an *and* operator, so an object must match both of them in order to be expired?

Reply:
the initial `0` is due to typescript enforcing the field.